File: Emulator/src/main/core/plugin/PluginManifest.java
package core.plugin;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Represents a plugin manifest.
 * @author Emperor
 */
@Retention(RetentionPolicy.RUNTIME)
public @interface PluginManifest {

    /**
     * Gets the plugin type.
     * @return The plugin type.
     */
    PluginType type() default PluginType.ACTION;

    /**
     * Get name.
     * @return the string
     */
    String name() default "";
}

File: game-server/src/main/core/plugin/PluginManifest.kt
package core.plugin

/**
 * Annotation for marking plugin metadata.
 *
 * @property type The plugin type.
 * @property name The plugin name.
 */
@Retention(AnnotationRetention.RUNTIME)
annotation class PluginManifest(
    val type: PluginType = PluginType.ACTION,
    val name: String = ""
)